sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java

```diff
@@ -819,19 +819,82 @@
         };
     }
 
-    private Function<PollingContext<PhoneNumberSearch>,
-        Mono<PhoneNumberSearch>> createSearchFetchResultOperation() {
+    private Function<PollingContext<PhoneNumberReservation>,
+        Mono<PhoneNumberReservation>> createSearchFetchResultOperation() {
         return pollingContext -> {
             return Mono.just(pollingContext.getLatestResponse().getValue());
         };
     }
+
+    /**
+     * LongRunningOperation for the purchaseSearch.
+     *
+     * @param reservationId ID of the reservation
+     * @param pollInterval Time lapse of the poll request.
+     * @return A {@link PollerFlux} object.
+     */
+
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PollerFlux<Void, Void> beginPurchaseReservation(String reservationId, Duration pollInterval) {
+        Objects.requireNonNull(reservationId, "'reservationId' can not be null.");
+        Objects.requireNonNull(pollInterval, "'pollInterval' can not be null.");
+
+        return new PollerFlux<Void, Void>(pollInterval,
+            purchaseReservationActivationOperation(reservationId),
+            purchaseReservationCreatePollOperation(reservationId),
+            (activationResponse, pollingContext) -> Mono.error(new RuntimeException("Cancellation is not supported")),
+            purchaseReservationFetchResultOperation());
+    }
+
+    private Function<PollingContext<Void>, Mono<Void>> purchaseReservationActivationOperation(String reservationId) {
+
+        return (pollingContext) -> {
+            Mono<Void> response = purchaseSearchWithResponse(reservationId).map(
+                purchaseResponse -> purchaseResponse.getValue());
+            return response;
+        };
+    }
+
+    private Function<PollingContext<Void>, Mono<PollResponse<Void>>>
+                purchaseReservationCreatePollOperation(String reservationId) {
+        return (pollingContext) -> getSearchById(reservationId)
+            .flatMap(getSearchResponse -> {
+                LongRunningOperationStatus status = null;
+                SearchStatus statusResponse = getSearchResponse.getStatus();
+                if (statusResponse.equals(SearchStatus.RESERVED) || statusResponse.equals(SearchStatus.EXPIRED)
+                    || statusResponse.equals(SearchStatus.SUCCESS)) {
+                    status = LongRunningOperationStatus.SUCCESSFULLY_COMPLETED;
+                } else if (statusResponse.equals(SearchStatus.ERROR)) {
+                    status = LongRunningOperationStatus.FAILED;
+                } else {
+                    status = LongRunningOperationStatus.IN_PROGRESS;
+                }
+
+                return Mono.just(new PollResponse<>(status, null));
+            });
+
+    }
+
+
+    private Function<PollingContext<Void>,
+        Mono<Void>> purchaseReservationFetchResultOperation() {
+
+        return pollingContext -> {
+
+            return Mono.just(pollingContext.getLatestResponse().getValue());
```

Comment:
What does the value contain? Isnt that the 'SearchStatus', in that case shouldnt you change the return type of this method and for the flux (its currently void)?

Reply:
She's returning a null in the pollOperation so this is fine, I guess

Reply:
ohh yea, you are right 👍